pkg/apis/pipeline/v1beta1/resultref.go

```diff
@@ -122,12 +133,21 @@ func stripVarSubExpression(expression string) string {
 	return strings.TrimSuffix(strings.TrimPrefix(expression, "$("), ")")
 }
 
-func parseExpression(substitutionExpression string) (string, string, error) {
+func parseExpression(substitutionExpression string) (string, string, string, error) {
 	subExpressions := strings.Split(substitutionExpression, ".")
-	if len(subExpressions) != 4 || subExpressions[0] != ResultTaskPart || subExpressions[2] != ResultResultPart {
-		return "", "", fmt.Errorf("Must be of the form %q", resultExpressionFormat)
+
+	// TODO (@yongxuanzhang): consider array type result here tasks.<taskName>.results.<resultName>[*] and indexing tasks.<taskName>.results.<resultName>[0]
```

Comment:
could you please link to the issue rather than a contributor? This provides a lot more context for anyone reading this comment.

Reply:
You can link to https://github.com/tektoncd/pipeline/issues/4723 if you want

Reply:
Let me just remove the todo line since @Yongxuanzhang has implemented this (I guess that will be merged soon) .

Originally I added this just to avoid people's confusion (i.e. why only string result and object result are considered here not array type? 😆

Reply:
Oh it is not opened yet, but this is a good reminder for me when I open that PR. We can keep it and link to our issue.